//------------------------------------------------------------------------------
// @file: Group.proto
// @author: Fabio Luchetti - CERN
//------------------------------------------------------------------------------

syntax="proto3";
package eos.console;

message IoProto {

  message StatProto {
    bool details      = 1;
    bool monitoring   = 2;
    bool numerical    = 3;
    bool top          = 4;
    bool domain       = 5;
    bool apps         = 6;
    bool summary      = 7;
    bool sample_stat   = 8;
    uint64 time_ago  = 9;
    uint64 time_interval = 10;
  }

  message EnableProto {
    bool switchx        = 1; // enable or disable
    bool reports        = 2;
    bool popularity     = 3;
    bool namespacex     = 4;
    string upd_address  = 5;
  }

  message ReportProto {
    string path = 1;
  }

  message NsProto {
    bool monitoring       = 1;
    bool rank_by_byte     = 2; // @note enum with by_access?
    bool rank_by_access   = 3;
    bool last_week        = 4;
    bool hotfiles         = 5;
    enum Count {
      NONE          = 0;
      ONEHUNDRED    = 1;
      ONETHOUSAND   = 2;
      TENTHOUSAND   = 3;
      ALL           = 4;
    }
    Count count           = 6;
  }

  message LimitProto {
    bool listing        = 1;
    string typefilter   = 2;
    string rangefilter  = 3;
    string keyfilter    = 4;
  }

  oneof subcmd {
    StatProto stat        = 1;
    EnableProto enable    = 2;
    ReportProto report    = 3;
    NsProto ns            = 4;
    LimitProto limit      = 5;
 }
}
